### PR TITLE
Yuhsuan/1372 colorbar preference

### PR DESCRIPTION
**Pull request by YuHsuan-Hwang**
This PR is for #1372: adding options to the preference dialog.

I am not sure if we should make the settings implicit:
we are only adding some of the settings to the preference dialog. If we make them implicit, some settings in the image view settings will remain when reopening CARTA but some will not. Users may not easily predict this behavior and may take time to be used to it.
I think it’s hard to decide if the user need a setting remain or not because it’s different from person to person and from case to case. It is more simple and clear to me to just use the preference dialog when necessary.

### CODE REVIEW

**Comment by veggiesaurus** [comment]
I think its fine to leave the implicit preference stuff to another feature, as we will need to be consistent across a number of image view settings

**Comment by YuHsuan-Hwang** [comment]
> I think its fine to leave the implicit preference stuff to another feature, as we will need to be consistent across a number of image view settings

@veggiesaurus Ok. I think we can leave it to #1169

**Comment by YuHsuan-Hwang** [comment]
@veggiesaurus @kswang1029 @duidae Changed to formal PR:
* added visible, interactive, position, width, ticks density, and label visible to the preference dialog
* same behavior with AST preference settings: apply the settings when opening / appending images

**Review by kswang1029 — APPROVED** [review]
this works fine as far as I can tell. 👍

File: src/components/Dialogs/PreferenceDialog/PreferenceDialogComponent.tsx [hunk 1]
```diff
@@ -317,6 +317,47 @@ export class PreferenceDialogComponent extends React.Component {
                         onChange={(event: React.FormEvent<HTMLSelectElement>) => preference.setPreference(PreferenceKeys.WCS_OVERLAY_WCS_TYPE, event.currentTarget.value)}
                     />
                 </FormGroup>
+                <FormGroup inline={true} label="Colorbar Visible">
+                    <Switch checked={preference.colorbarVisible} onChange={ev => preference.setPreference(PreferenceKeys.WCS_OVERLAY_COLORBAR_VISIBLE, ev.currentTarget.checked)} />
+                </FormGroup>
+                <FormGroup inline={true} label="Colorbar Interactive">
+                    <Switch checked={preference.colorbarInteractive} onChange={ev => preference.setPreference(PreferenceKeys.WCS_OVERLAY_COLORBAR_INTERACTIVE, ev.currentTarget.checked)} />
+                </FormGroup>
+                <FormGroup inline={true} label="Position">
```
Comment (veggiesaurus) [inline src/components/Dialogs/PreferenceDialog/PreferenceDialogComponent.tsx]: To be consistent with the other labels, should this not be "Colorbar Position"?

**Review by veggiesaurus — CHANGES_REQUESTED** [review]

File: src/components/Dialogs/PreferenceDialog/PreferenceDialogComponent.tsx (hunk 1 above)
Comment (YuHsuan-Hwang) [reply src/components/Dialogs/PreferenceDialog/PreferenceDialogComponent.tsx]: @veggiesaurus Yes, thanks. This is changed.

**Pull request merged**